testament/categories.nim

```diff
@@ -499,20 +508,25 @@ proc testNimblePackages(r: var TResults; cat: Category; packageFilter: string, p
       var test = makeSupTest(url, "", cat)
       let buildPath = packagesDir / name
       if not existsDir(buildPath):
-        if hasDep:
-          let installName = if url.len != 0: url else: name
-          var message: string
-          if not actionRetry(maxRetry = 3, backoffDuration = 1.0,
-            (proc(): bool = nimbleInstall(installName, message))):
-            r.addResult(test, targetC, "", message, reInstallFailed)
-            continue
-
         let (installCmdLine, installOutput, installStatus) = execCmdEx2("git", ["clone", url, buildPath])
         if installStatus != QuitSuccess:
           let message = "git clone failed:\n$ " & installCmdLine & "\n" & installOutput
           r.addResult(test, targetC, "", message, reInstallFailed)
           continue
 
+        if hasDep:
+          var message: string
+          if not actionRetry(maxRetry = 3, backoffDuration = 1.0,
+            proc: bool =
+               let (outp, status) = execCmdEx("nimble install -y", workingDir = buildPath)
```

Comment:
* this installs (and then tests) from git HEAD, which is less stable than latest release tag, so may break more; at least it addresses inconsistency part of https://github.com/timotheecour/Nim/issues/167
* so we want `nimble install` or `nimble develop` here? this matters when resolving references to a package name

Reply:
Yes, for now we test git HEAD. I plan on adding an option to use `nimble develop` instead.

Reply:
Added an option to either test git HEAD or latest nimble release for now.